hedera-mirror-rest/entity-info-updater/utils.js

```
@@ -0,0 +1,181 @@
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+'use strict';
+
+// external libraries
+
+const proto = require('@hashgraph/proto');
+const {KeyList, PublicKey} = require('@hashgraph/sdk');
+const _ = require('lodash');
+const fs = require('fs');
+const log4js = require('log4js');
+const math = require('mathjs');
+
+// local
+const config = require('./config');
+
+const logger = log4js.getLogger();
+const longMinValue = -9223372036854775808;
```

This is silently truncating these numbers to the max Number. You need to use BigInt by adding `n`

---

This was meant to match the java logic that handles the overflow. Will revisit

---

Gotcha. Will use Bigint here also

---

BigInt used